membership-attribute-service/app/components/TouchpointBackends.scala

```diff
@@ -1,27 +1,56 @@
 package components
 
 import akka.actor.ActorSystem
+import com.gu.memsub.subsv2.services.{CatalogService, SubscriptionService}
+import com.gu.salesforce.SimpleContactRepository
+import com.gu.zuora.ZuoraSoapService
+import com.gu.zuora.rest.ZuoraRestService
 import com.typesafe.config.Config
 import configuration.Stage
 import monitoring.CreateMetrics
-import services.SupporterProductDataService
+import services.{HealthCheckableService, SupporterProductDataService}
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
 class TouchpointBackends(
     actorSystem: ActorSystem,
     config: Config,
     createMetrics: CreateMetrics,
-    supporterProductDataServiceOverride: Option[SupporterProductDataService],
+    supporterProductDataServiceOverride: Option[SupporterProductDataService] = None,
+    contactRepositoryOverride: Option[SimpleContactRepository] = None,
+    subscriptionServiceOverride: Option[SubscriptionService[Future]] = None,
+    zuoraRestServiceOverride: Option[ZuoraRestService[Future]] = None,
+    catalogServiceOverride: Option[CatalogService[Future]] = None,
+    zuoraServiceOverride: Option[ZuoraSoapService with HealthCheckableService] = None,
 )(implicit
     executionContext: ExecutionContext,
 ) {
   private val defaultStage = Stage(config.getString("touchpoint.backend.default"))
-  val normal = new TouchpointComponents(defaultStage, createMetrics, config, supporterProductDataServiceOverride)(
+  val normal = new TouchpointComponents(
+    defaultStage,
+    createMetrics,
+    config,
+    supporterProductDataServiceOverride,
+    contactRepositoryOverride,
+    subscriptionServiceOverride,
+    zuoraRestServiceOverride,
+    catalogServiceOverride,
+    zuoraServiceOverride,
+  )(
     actorSystem,
     executionContext,
   )
   private val testStage = Stage(config.getString("touchpoint.backend.test"))
   val test =
-    new TouchpointComponents(testStage, createMetrics, config, supporterProductDataServiceOverride)(actorSystem, executionContext)
+    new TouchpointComponents(
+      testStage,
```

Comment:
This is a bit strange now because we're using the same services for normal and test which defeats the point of having the two sets of components

Reply:
If it was up to me, I'd throw away the whole test user code path.
Which is not that hard, cause it's just deleting code.

Reply:
If we ever get to testing the test users code path (which is probably never), we can address this.
They are different cases though - acceptance / integration testing and test users.

Reply:
Fair enough